File: 3668-restore-finishing-order/3668-restore-finishing-order.ts
function recoverOrder(order: number[], friends: number[]): number[] {
    let result =[]
    let friendSet =new Set(friends)
    for(let i=0;i<order.length;i++){
        if(friendSet.has(order[i])){
            result.push(order[i])
        }
    }
    return result
};

File: 3668-restore-finishing-order/3668-restore-finishing-order.cs
public class Solution {
    public int[] RecoverOrder(int[] order, int[] friends) {
        List<int> result = new List<int>();
        for(int i=0;i<order.Length;i++){
            for(int j=0;j<friends.Length;j++){
                if(order[i]==friends[j]){
                   result.Add(order[i]);
                }
            }
        }
        return result.ToArray();
    }
}